toolkit/tests/service_client/test_client.py

```diff
@@ -0,0 +1,39 @@
+from dataclasses import dataclass
+from typing import Any, Optional
+from unittest.mock import patch
+
+from pythonit_toolkit.api.service_client import ServiceClient
+from ward import test
+
+
+@dataclass
+class MockResponse:
+    return_value: Optional[Any] = None
+
+    async def json(self):
+        return self.return_value
+
+
+@test("execute a query")
+async def _():
+    query = """
+        query{
+            users {
+                id
+            }
+        }
+    """
+    mock_response = {"data": {"users": [{"id": 1}]}}
+
+    with patch("httpx.AsyncClient.post") as post_mock:
+        post_mock.return_value = MockResponse(return_value=mock_response)
```

Comment:
something like this should work so you don't need the MockResponse dataclass? mocks usually are very flexible

```suggestion
        from unittest.mock import Mock
        post_mock.return_value = Mock()
        post_mock.return_value.json.return_value = mock_response
```

Reply:
but json is async, that is why I made a custom class... I wasn't able to make it work with a normal Mock object... :(

Reply:
Uhh there is also AsyncMock!